drivers/volume/portworx/portworx.go

```diff
@@ -2834,13 +2834,13 @@ func (p *portworx) GetBackupStatus(backup *storkapi.ApplicationBackup) ([]*stork
 				if st.Code() == codes.Unimplemented {
 					// if using old porx version, return 0 size
 					vInfo.Size = 0
-					volumeInfos = append(volumeInfos, vInfo)
-					return volumeInfos, nil
+					logrus.Warnf("Unsupported porx version to fetch backup size for backup ID %v: %v", csStatus.cloudSnapID, err)
+				} else {
+					logrus.Errorf("Failed to fetch backup size for backup ID %v: %v", csStatus.cloudSnapID, err)
```

Comment:
If the PORX supports size API but still if we got some error, what should be the behaviour?
Do we need to return error immediately from  GetBackupStatus  or just print the error message and make the size of that volume to be zero?

Reply:
My understanding is to continue as there could be a case where each volume can be on a different provider which might not fail.

Reply:
1) If we continue, we might end up in giving wrong size information for those volumes for which size API failed. It might be confusing to the user that backup was success but size is zero.
2) Also, Do we need to make vInfo.Size to zero? Will it not default be zero. If not, we might need to do that in the error case as well.

Reply:
Initially, I had it to return an error and then changed it.  I will rever it to return an error that will fail the backup.